lib/jwe/header.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module JWE
+  # Generates JWE header from algorithm, encryption, and compression parameters
+  class Header
```

Comment:
As there is no state I would prefer module and static methods over a class that needs instantiation in this case.